src/py/flwr/driver/driver.py

```diff
@@ -126,3 +128,77 @@ def pull_task_res(self, req: PullTaskResRequest) -> PullTaskResResponse:
         # Call Driver API
         res: PullTaskResResponse = self.stub.PullTaskRes(request=req)
         return res
+
+
+class Driver:
+    """`Driver` class provides an interface to the Driver API."""
+
+    def __init__(self) -> None:
+        self.grpc_driver = GrpcDriver()
+        self.workload_id = 0
+        self.task_id_pool: Set[str] = set()
+        self.node = Node(node_id=0, anonymous=True)
+
+    def _check_and_init_grpc_driver(self) -> None:
+        # Check if the GrpcDriver is initialized
+        if self.workload_id != 0:
+            return
+
+        # Connect and create workload
+        self.grpc_driver.connect()
+        res = self.grpc_driver.create_workload(CreateWorkloadRequest())
+        self.workload_id = res.workload_id
+
+    def get_nodes(self) -> List[Node]:
+        """Get client IDs."""
+        self._check_and_init_grpc_driver()
+
+        # Call GrpcDriver method
+        res = self.grpc_driver.get_nodes(GetNodesRequest(workload_id=self.workload_id))
+        return list(res.nodes)
+
+    def push_task_ins(self, task_ins_list: Iterable[TaskIns]) -> List[str]:
+        """Schedule tasks."""
+        self._check_and_init_grpc_driver()
+
+        # Set workload_id
+        for task_ins in task_ins_list:
+            task_ins.workload_id = self.workload_id
+
+        # Call GrpcDriver method
+        res = self.grpc_driver.push_task_ins(
+            PushTaskInsRequest(task_ins_list=task_ins_list)
+        )
+
+        # Cache received task_ids
+        self.task_id_pool.update(res.task_ids)
+        return list(res.task_ids)
+
+    def pull_task_res(self, task_ids: Optional[Iterable[str]] = None) -> List[TaskRes]:
+        """Get task results.
+
+        Retrieve all task results if `task_ids` is None.
+        """
+        self._check_and_init_grpc_driver()
+
+        # Check if task_ids is None
+        if task_ids is None:
+            task_ids = list(self.task_id_pool)
+
+        # Call GrpcDriver method
+        res = self.grpc_driver.pull_task_res(
+            PullTaskResRequest(node=self.node, task_ids=task_ids)
+        )
```

Comment:
This can fail

Reply:
As discussed, we will add exception handling mechanism in a following PR.